### PR TITLE
refactor(pc): string -> [4]bytes

### PR DESCRIPTION
**Pull request by itsdevbear**
- More performant
- Remove `unsafe` and `reflect` usage

### CODE REVIEW

**Comment by itsdevbear** [comment]
todo: write a quick test to make sure it's not using the pointer memory address as the key by accident

File: eth/core/precompile/stateful_container.go [hunk 1]
```diff
@@ -42,7 +41,7 @@ type statefulContainer struct {
 	// method signatures) to native precompile functions. The signature key is provided by the
 	// precompile creator and must exactly match the signature in the geth abi.Method.Sig field
 	// (geth abi format). Please check core/precompile/container/method.go for more information.
-	idsToMethods map[string]*method
+	idsToMethods map[[4]byte]*method
```
Comment (calbera) [inline eth/core/precompile/stateful_container.go:44]: i would make a `type methodSelector [4]byte` or smth

Comment (calbera) [reply eth/core/precompile/stateful_container.go:44]: `type methodSelector [NumBytesMethodID]byte`

File: eth/core/precompile/method.go [hunk 2]
```diff
@@ -31,6 +31,9 @@ import (
 	"pkg.berachain.dev/polaris/lib/utils"
 )
 
+// methodID is a fixed length byte array that represents the method ID of a precompile method.
+type methodID = [NumBytesMethodID]byte
```
Comment (calbera) [inline eth/core/precompile/method.go:35]: nit but I think this should be a type alias rather than a type declaration

Comment (itsdevbear) [reply eth/core/precompile/method.go:35]: sure

**Review by calbera — APPROVED** [review]

**Pull request merged**